dev/devicelab/lib/framework/running_processes.dart

```diff
@@ -0,0 +1,228 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+@immutable
+class RunningProcessInfo {
+  const RunningProcessInfo(this.pid, this.creationDate, this.commandLine)
+      : assert(pid != null),
+        assert(commandLine != null);
+
+  final String commandLine;
+  final int pid;
+  final DateTime creationDate;
+
+  @override
+  bool operator ==(Object other) {
+    return other is RunningProcessInfo && other.pid == pid;
+  }
+
+  @override
+  int get hashCode => pid.hashCode;
+
+  @override
+  String toString() {
+    return 'RunningProcesses{pid: $pid, commandLine: $commandLine, creationDate: $creationDate}';
+  }
+}
+
+Future<bool> killProcess(int pid, {ProcessManager processManager}) async {
+  assert(pid != null, 'Must specify a pid to kill');
+  processManager ??= const LocalProcessManager();
+  ProcessResult result;
+  if (Platform.isWindows) {
+    result = await processManager.run(<String>[
+      'taskkill',
+      '/pid', pid.toString(),
+      '/f',
+    ]);
+  } else {
+    result = await processManager.run(<String>[
+      'kill',
+      '-9',
+      pid.toString(),
+    ]);
+  }
+  return result.exitCode == 0;
+}
+
+Stream<RunningProcessInfo> getRunningProcesses({
+  String processName,
+  ProcessManager processManager,
+}) {
+  processManager ??= const LocalProcessManager();
```

Comment:
Alternatively, could expose this as a default argument. That won't get set if `null` is explicitly provided though

Reply:
Do we have asserts enabled in this?

Reply:
I've added some `continue;`s in this case, as this could just be output we don't care about.

Reply:
No asserts, but you can still use required.

Reply:
It can't be required if a default is specified